src/substrait/textplan/converter/InitialPlanProtoVisitor.h

```diff
@@ -56,6 +57,28 @@ class InitialPlanProtoVisitor : public BasePlanProtoVisitor {
   std::any visitNamedStruct(
       const ::substrait::proto::NamedStruct& named) override;
 
+  void updateLocalSchema(
+      const std::shared_ptr<RelationData>& relationData,
+      const ::substrait::proto::Rel& relation);
+
+  void addFieldsToRelation(
+      const std::shared_ptr<RelationData>& relationData,
+      const ::substrait::proto::Rel& relation);
+
+  void addFieldsToRelation(
+      const std::shared_ptr<RelationData>& relationData,
+      const ::substrait::proto::Rel& left,
+      const ::substrait::proto::Rel& right);
+
+  template <typename T>
+  void addFieldsToRelation(
+      const std::shared_ptr<RelationData>& relationData,
+      const T& relations) {
+    for (const auto& rel : relations) {
+      addFieldsToRelation(relationData, rel);
+    }
+  };
```

Comment:
This is a little generic.  I'm not sure there is much we can do about it.  Ideally we'd be able to restrict it to "things that can be ranged over" but without concepts (e.g. https://en.cppreference.com/w/cpp/ranges/range) I don't think we can easily do that.

Reply:
Agreed, I did look at ranges when I was implementing this.